src/EnergyPlus/AirflowNetwork/src/Solver.cpp

```diff
@@ -12289,6 +12294,74 @@ namespace AirflowNetwork {
+
+        m_state.dataInputProcessing->inputProcessor->getObjectDefMaxArgs(m_state, currentModuleObject, TotalArgs, NumAlphas, NumNumbers);
+
+        Alphas.allocate(NumAlphas);
+        cAlphaFields.allocate(NumAlphas);
+        cNumericFields.allocate(NumNumbers);
+        Numbers.dimension(NumNumbers, 0.0);
+        lAlphaBlanks.dimension(NumAlphas, true);
+        lNumericBlanks.dimension(NumNumbers, true);
+
+        for (int linkageNum = 1; linkageNum <= numObjects; ++linkageNum) {
+            m_state.dataInputProcessing->inputProcessor->getObjectItem(m_state,
+                                                                       currentModuleObject,
+                                                                       linkageNum,
+                                                                       Alphas,
+                                                                       NumAlphas,
+                                                                       Numbers,
+                                                                       NumNumbers,
+                                                                       IOStatus,
+                                                                       lNumericBlanks,
+                                                                       lAlphaBlanks,
+                                                                       cAlphaFields,
+                                                                       cNumericFields);
+            // check Alphas(2) and Alphas(3) (node names) to see what the
+            bool validFirstLinkage2 = false;
+            for (int nodeNum = 1; nodeNum <= numNodes; ++nodeNum) {
+                if (Util::SameString(Alphas(2), DisSysNodeData(nodeNum).Name)) { // found the match
+                    if (!Util::SameString(DisSysNodeData(nodeNum).EPlusType, "Other") && !Util::SameString(DisSysNodeData(nodeNum).EPlusType, "")) {
+                        validFirstLinkage2 = true;
+                    }
+                    break;
+                }
+            }
+            bool validFirstLinkage3 = false;
+            for (int nodeNum = 1; nodeNum <= numNodes; ++nodeNum) {
+                if (Util::SameString(Alphas(3), DisSysNodeData(nodeNum).Name)) { // found the match
+                    if (!Util::SameString(DisSysNodeData(nodeNum).EPlusType, "Other") && !Util::SameString(DisSysNodeData(nodeNum).EPlusType, "")) {
+                        validFirstLinkage3 = true;
+                    }
+                    break;
+                }
+            }
+            if (validFirstLinkage2 || validFirstLinkage3) {
+                shiftResult = linkageNum;
+                break;
+            }
+            if (!lAlphaBlanks(5)) { // There might be a valid zone here which will also make this a "first" linkage in the input
+                shiftResult = linkageNum;
+                break;
+            }
+        }
+        return shiftResult;
+    }
```

Comment:
This just looks like it's finding the first AirflowNetwork:Distribution:Linkage that matches one of the AirflowNetwork:Distribution:Node objects and then reading the AirflowNetwork:Distribution:Linkage objects in order. I don't know how that aligns these objects in some manner or how this works for other random orders of AirflowNetwork:Distribution:Linkage objects so I can't comment more. I would ask that @jasondegraw look at this.